arangod/IResearch/IResearchViewBlock.cpp

```diff
@@ -48,6 +48,47 @@
 #include "search/boolean_filter.hpp"
 #include "search/score.hpp"
 
+namespace {
+
+typedef std::vector<arangodb::iresearch::DocumentPrimaryKey> pks_t;
+
+constexpr size_t const BATCH_SIZE = 1024;
+
+pks_t::iterator readPKs(
+    irs::doc_iterator& it,
+    irs::columnstore_reader::values_reader_f const& values,
+    pks_t& keys,
+    size_t limit
+) {
+  keys.clear();
+  keys.resize(limit);
+
+  auto begin = keys.begin();
+  auto end = keys.end();
+
+  for (irs::bytes_ref key; begin != end && it.next(); ) {
```

Comment:
Is `it.next()` moving the iterator forward already? If so, doesn't this skip over the first item?

Reply:
iresearch iterators are "before the first element" after creation, because getting even first element might be expensive and useless in some cases (e.g. disjunction)

Reply:
vale